webroot/panel/account.php

```diff
@@ -130,23 +135,69 @@
 
 <hr>
 
+<form action="" method="POST">
+
+    <input type="hidden" name="form_type" value="loginshell">
+
+    <select id="loginSelector" name= "shellSelect"> 
+
+        <option value="" disabled hidden>Select Login Shell...</option>
+
+        <?php
+        $cur_shell = $USER->getLoginShell();
+        $found_selector = false;
+        foreach ($BRANDING["loginshell"]["shell"] as $shell) {
+            if ($cur_shell == $shell){
+                echo "<option selected>$shell</option>";
+                $found_selector = true;
+            } else {
+                echo "<option>$shell</option>";
+            }
+        }
+
+        if ($found_selector) {
+            echo "<option value = 'custom'> Custom </option>";
+        } else {
+            echo "<option value = 'custom' selected> Custom </option>";
+        }
+        ?>
+
+        
+        
+    </select>
 <?php
-echo "<h5>Login Shell</h5>";
-echo
-"<div class='inline'>
-<form action='' method='POST'>
-<input type='hidden' name='form_type' value='loginshell'>
-<input type='text' name='loginshell' placeholder='Login Shell (ie. /bin/bash)' 
-value=" . $USER->getLoginShell() . " required>
-<input type='submit' value='Set Login Shell'>
-</form>
-</div>";
+
+    if ($found_selector) {
+        echo "<input id='customLoginBox' type='text' name='shell'>";
+    } else {
+        echo "<input id='customLoginBox' type='text' name='shell' value= '$cur_shell'>";
+    }
 ?>
+    <input type='submit' value='Set Login Shell'>
+
+</form>
+
+
 
 <script>
     $("button.btnAddKey").click(function() {
         openModal("Add New Key", "<?php echo $CONFIG["site"]["prefix"]; ?>/panel/modal/new_key.php");
     });
+
+    $("#customLoginBox").hide();
```

Comment:
This is what I was talking about during our meeting - in the javascript you only have to hide the box **if** the user has a custom shell currently selected.

This block is untested, make sure it runs in the dev environment. If the user has a custom shell the text box should appear on page refresh.

```suggestion
    var customLoginBox = $("#customLoginBox");
    if (customLoginBox.val() == "") {
        // login box is empty, so we hide it by default
        // if the login box had a value, that means it would be a custom shell
        // and should not hide by default
        customLoginBox.hide();
    }
```